ERCS/erc-7677.md

```diff
@@ -0,0 +1,296 @@
+
+## Specification
+
+One new [EIP-5792](./eip-5792.md) wallet capability is defined. We also define a standard interface for paymaster web services as a prerequisite.
+
+### Paymaster Web Service Interface
+
+We define two JSON-RPC methods to be implemented by paymaster web services.
+
+#### `pm_getPaymasterStubData`
+
+Returns stub values to be used in paymaster-related fields of an unsigned user operation for gas estimation. Accepts an unsigned user operation, entrypoint address, chain id, and a context object. Paymaster service providers can define fields that app developers should use in the context object. 
+
+##### `pm_getPaymasterStubData` RPC Specification
+
+```typescript
+// [userOp, entryPoint, chainId, context]
+type GetPaymasterStubDataParams = [
+  // Below is specific to Entrypoint v0.6 but this API can be used with other entrypoint versions too
+  {
+    sender: `0x${string}`;
+    nonce: `0x${string}`;
+    initCode: `0x${string}`;
+    callData: `0x${string}`;
+    callGasLimit: `0x${string}`;
+    verificationGasLimit: `0x${string}`;
+    preVerificationGas: `0x${string}`;
+    maxFeePerGas: `0x${string}`;
+    maxPriorityFeePerGas: `0x${string}`;
+  },
+  `0x${string}`,
+  `0x${string}`,
+  Record<string, any>
+];
+
+type GetPaymasterStubDataResult = Record<string, `0x${string}`> // Stub paymaster values
+```
+
+###### `pm_getPaymasterStubData` Example Parameters
+
+```json
+[
+  {
+    "sender": "0x...",
+    "nonce": "0x...",
+    "initCode": "0x",
+    "callData": "0x...",
+    "callGasLimit": "0x...",
+    "verificationGasLimit": "0x...",
+    "preVerificationGas": "0x...",
+    "maxFeePerGas": "0x...",
+    "maxPriorityFeePerGas": "0x...",
+    "paymasterAndData": "0x...",
+    "signature": "0x..."
+  },
+  "0x5FF137D4b0FDCD49DcA30c7CF57E578a026d2789",
+  "0x2105",
+  {
+    // Illustrative context field. These should be defined by service providers.
+    "policyId": "962b252c-a726-4a37-8d86-333ce0a07299"
+  }
+]
+```
+
+###### `pm_getPaymasterStubData` Example Return Value
+
+Paymaster services MUST detect which entrypoint version the account is using and return the correct fields.
+
+For example, if using entrypoint v0.6:
+
+```json
+{
+  "paymasterAndData": "0x..."
+}
+```
+
+If using entrypoint v0.7:
+
+```json
+{
+  "paymaster": "0x...",
+  "paymasterData": "0x..."
+}
+```
+
+#### `pm_getPaymasterData`
+
+Returns values to be used in paymaster-related fields of a signed user operation. These are not stub values and will be used during user operation submission to a bundler. Similar to `pm_getPaymasterStubData`, accepts an unsigned user operation, entrypoint address, chain id, and a context object.
+
+This method also returns a `sponsor` object with `name` and `icon` fields. The `name` field is the name of the party sponsoring the transaction, and the `icon` field is a URI pointing to an image. Wallet developers MAY choose to display sponsor information to users.
```

Comment:
It might be worth sticking some normative requirements in here to keep the URI field from being a JS injection vector, of [the sort that some reviewer (apologies I can't remember which) inserted into ERC-6963 at the last minute](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-6963.md?plain=1#L76).  Alternately, you could mention this URI security in security considerations, but I recommend doing it normatively here in the very section that every [wallet] implementer will read.

Reply:
added the same note here.